sql-to-dbsp-compiler/lib/sqllib/src/variant.rs

```diff
@@ -387,18 +551,240 @@ where
     }
 }
 
-#[cfg(test)]
-mod test {
-    use super::Variant;
-    use dbsp::RootCircuit;
+//////////////////// Reverse conversions Variant -> T
 
-    #[test]
-    fn circuit_accepts_variant() {
-        let (_circuit, _input_handle) = RootCircuit::build(move |circuit| {
-            let (_stream, input_handle) = circuit.add_input_zset::<Variant>();
-            Ok(input_handle)
-        })
-        .unwrap();
+#[derive(Error, Debug)]
+pub enum ConversionError {
+    #[error("{0}")]
+    CustomError(String),
+}
+
+impl ConversionError {
+    pub fn from_string(message: String) -> Box<Self> {
+        Box::new(ConversionError::CustomError(message))
+    }
+
+    pub fn from_strng(message: &str) -> Box<Self> {
+        Box::new(ConversionError::CustomError(message.to_string()))
+    }
+}
+
+macro_rules! into {
+    ($variant:ident, $type:ty) => {
+        impl TryFrom<Variant> for $type {
+            type Error = Box<dyn Error>;
```

Comment:
probably shouldn't have to Box the error?

Reply:
I spent one hour trying to make the errors work. Can you show me an example?

Reply:
I can try to fix it tomorrow